custom/abdm/milestone_one/views/abha_verification_views.py

```diff
@@ -22,6 +24,9 @@ def get_auth_methods(request):
     if not health_id:
         error_msg = "Missing required parameter: health_id"
         return generate_invalid_req_response(error_msg)
+    if check_for_existing_abha_number(request.user.domain, health_id):
+        error_msg = "Provided ABHA is already linked to another beneficiary!"
+        return generate_invalid_req_response(error_msg, response_code=HTTP_409_CONFLICT)
```

Comment:
nit: thinking if using 409 http code is an overkill here.

Additionally, I believe 409 is used for update conflicts?
A generic 400 seems equally good here.

Reply:
> This is actually the code inside the response body, the status code would still be 400.

Ya, I don't think mixing http status codes within the response would be a nice thing.

If we want to communicate specifically that abha was already present in the system, we could something specific in the response, other than the message to be shown to the user, that specifically is for this scenario, like an error_code, used and agreed between HQ and ABDM app, so the response code have, `"error_code": "abha_already_in_use"` or something better

Reply:
> Ya, I don't think mixing http status codes within the response would be a nice thing.
Agreed and thanks for pointing this out. 
I  believe the `code` here is the error code itself . We should use numbers (non HTTP codes) or even string as suggested. ABDM Gateway also requires some specific format of error for M2/M3 for API's called by them/us.
`{
  "error": {
    "code": 1000,
    "message": "string"
  }
}`

Will go ahead and use '1001' for this error case.
For now, I am refraining from making changes in the error schema to avoid any potential change on the mobile app side.
However, as part of M2 and M3 I am going to revisit this and see the best way to have a consistent schema.

Reply:
> Will go ahead and use '1001' for this error case.

where is 1001 coming from? is that from a list of codes?

>  However, as part of M2 and M3 I am going to revisit this and see the best way to have a consistent schema.

okay.

Reply:
> where is 1001 coming from? is that from a list of codes?

Yes created a dict of one error code for now [here](https://github.com/dimagi/commcare-hq/pull/33006/commits/bcfb407ef455b2a5292a272c243055debb5bf17d#diff-f05a4abebffab2730d016e780ba275ec4386a02173b5d30cdbe94116ad4c27cdR12)

Reply:
Beautiful